webapp/store/reducers/nav.js

```diff
@@ -0,0 +1,40 @@
+import { ADD_SIDE_NAV, REMOVE_SIDE_NAV } from '../constants/nav';
+import { initialMetadata } from '../../plugins/plugins';
+import { sortedNavItems } from '../selectors/nav';
+
+const pluginMetadata = initialMetadata();
+const initialState = {
+  sidebar: []
+};
+
+const navStore = (state = initialState, action) => {
+  const newState = { ...state };
+  const { type, payload } = action;
+
+  switch (type) {
+    case ADD_SIDE_NAV: {
```

Comment:
what do you think of:

`newState.sidebar = [...newState.sidebar, payload];`

Reply:
great idea!